taipy/core/job/_job_manager.py

```diff
@@ -50,9 +50,9 @@ def _create(
             force=force,
             version=version,
         )
-        cls._set(job)
         Notifier.publish(_make_event(job, EventOperation.CREATION))
         job._on_status_change(*callbacks)
+        cls._set(job)
```

Comment:
Due to failed tests that are checking for `Job`'s `_subscribers` in `tests/core/_orchestrator/test_orchestrator__submit.py`. Previously these tests passed because the jobs are taken from memory which contains the callbacks. But because the submit function now returns a Submission, the jobs are taken from `Submission.jobs` -> jobs are taken from repos -> tests failed because these callbacks are stored in memory for the moment (and yet in repos). So I make the changes to make sure the callbacks are available both in memory and repos of all time

Reply:
it does, though I'd avoid doing that to prevent further complicating the PR. Would suggest to do it in a separate PR since we're not sure of the side effect of doing that.